official/resnet/resnet.py

```diff
@@ -164,10 +168,59 @@ def building_block(inputs, filters, training, projection_shortcut, strides,
     strides: The block's stride. If greater than 1, this block will ultimately
       downsample the input.
     data_format: The input format ('channels_last' or 'channels_first').
+    version: The ordering convention for convolution, batch normalization, and
+             activation.
 
   Returns:
     The output tensor of the block.
   """
+  if version == 1:
+    return _building_block_v1(inputs=inputs, filters=filters, training=training,
+                              projection_shortcut=projection_shortcut,
+                              strides=strides, data_format=data_format)
+  if version == 2:
+    return _building_block_v2(inputs=inputs, filters=filters, training=training,
+                              projection_shortcut=projection_shortcut,
+                              strides=strides, data_format=data_format)
+  raise ValueError("version should be 1 or 2")
```

Comment:
THis check can also move to the Model.__init__. May as well be more specific/properly capitalized, too.

Reply:
Ditto.